cibuildwheel/linux.py

```diff
@@ -129,72 +123,31 @@ def build(project_dir, package_name, output_dir, test_command, test_requires, be
             gid=os.getgid(),
         )
 
-        container_name = 'cibuildwheel-{}'.format(uuid.uuid4())
+        def run_docker(command, stdin_str=None):
+            print('docker command: docker {}'.format(' '.join(map(shlex_quote, command))))
+            if stdin_str is None:
+                return subprocess.check_call(['docker'] + command)
+            else:
+                process = subprocess.Popen(['docker'] + command,
+                                           stdin=subprocess.PIPE, universal_newlines=True)
+                try:
+                    process.communicate(stdin_str)
+                except KeyboardInterrupt:
+                    process.kill()
+                    process.wait()
+                return process.returncode
```

Comment:
This function is a little strange in that if you give it a stdin_str, it doesn't raise on failure. I think maybe it would be better to raise a subprocess.CalledProcessError if the return code is non-zero, then change error handling at the call site

Reply:
True; let me change that.